orbit/pkg/update/update.go

```diff
@@ -515,6 +532,23 @@ func (u *Updater) checkExec(target, tmpPath string, customCheckExec func(execPat
 		return nil
 	}
 
+	platformGOARCH, err := goarchFromPlatform(localTarget.Info.Platform)
+	if err != nil {
+		return err
+	}
+	var containsArch bool
+	for _, arch := range platformGOARCH {
+		if arch == runtime.GOARCH {
+			containsArch = true
+		}
+	}
+	if !containsArch {
```

Comment:
Are you adding this because it's used by `fleetctl package`, right?
Any way that we can determine this is being executed in fleetctl vs real orbit?

Am worried in case we incorrectly add an arm64 executable to the amd64 target (or viceversa), which would permanently break the host if we don't do `checkExec`.

Reply:
There are a couple of ways we could do this. 

1. To detect the file type we could shell out to the `file` command and parse the output for the arch, although I'm not sure if the command output is consistent across platforms
2. Bring in a library that replicates the behaviour of `file` but written as a go library

Reply:
How about something like:
```go
if !containsArch && strings.HasSuffix(os.Args[0], "fleetctl") {
```
?

Reply:
That works as well, but it doesn't actually verify anything about the file

Reply:
But this is at package generation time so if there's a arch mismatch then it is under our control to fix, right?

Reply:
This is currently the case when generating MSI packages from linux, we don't run the `checkExec`.

Reply:
I'll go with your suggestion then